sklearn/neighbors/base.py

```diff
@@ -332,14 +336,25 @@ class from an array representing our data set and ask who's
                 raise ValueError(
                     "%s does not work with sparse matrices. Densify the data, "
                     "or set algorithm='brute'" % self._fit_method)
-            result = self._tree.query(X, n_neighbors,
-                                      return_distance=return_distance)
+            if n_jobs == 1:
```

Comment:
I wanted to "avoid any overhead". But you are right, it's absolutely not necessary.